### PR TITLE
Properly support Pokemon with no ability

### PR DESCRIPTION
**Pull request by MacChaeger**
As detailed here: https://www.smogon.com/forums/threads/pokemon-sun-moon-battle-mechanics-research.3586701/page-76#post-7790209

PS actually already supports this ability, but this adds the Trace interaction.

### CODE REVIEW

**Comment by Zarel** [comment]
I would rather not add a null entry to Abilities to achieve this. Is there another workaround for this?

**Comment by Zarel** [comment]
Can you tell me what this fixes? Maybe I can think of one.

**Comment by Zarel** [comment]
Oh, I see. I would call this "No Ability" just so its ID is clear.

**Comment by Slayer95** [comment]
We already have "None" in Gen 1 and 2. We might as well standardize on its name.
I am not a fan of any of ``""``, ``"None"`` and ``"No Ability"``, but "No Ability" sounds like a good compromise and it should be namespace-safe enough.

**Comment by Slayer95** [comment]
``mods/gen1/abilities.js`` and ``mods/gen2/abilities.js`` should be safe to delete now, too.

File: data/abilities.js [hunk 1]
```diff
@@ -40,6 +40,13 @@ Ratings and how they work:
 
 /**@type {{[k: string]: AbilityData}} */
 let BattleAbilities = {
+	"noability": {
+		shortDesc: "Does nothing.",
+		id: "noability",
+		name: "No Ability",
+		rating: 0.1,
+		num: 0,
```
Comment (Slayer95) [inline data/abilities.js]: This ability needs the ``isNonstandard`` flag

Comment (KrisXV) [reply data/abilities.js]: It’s an actual ability though

Comment (Slayer95) [reply data/abilities.js]: > do we expect this to be released eventually

The answer for that is no. So, it's not unreleased! It is, however, homologue to Missingno as an ability.

**Comment by Zarel** [comment]
Cool, thanks!

**Pull request merged**